htsget-storage/src/s3.rs

```diff
@@ -365,7 +365,7 @@ pub(crate) mod tests {
         )
         .await
         .unwrap();
-      assert!(result.url.starts_with("http://folder.localhost:8014/key2"));
+      assert!(result.url.starts_with("http://folder.localhost:0/key2"));
```

Comment:
Port 0?? Interesting!

Reply:
Yes, this allows allocating an arbitrary port so that it doesn't sometimes conflict with other instances of the same function (depending on concurrency). This is what was causing the tests to fail in CI but not locally: https://github.com/umccr/htsget-rs/actions/runs/11171864755/job/31478706837